src/Features/CSharp/Portable/CodeFixes/AddExplicitCast/AddExplicitCastCodeFixProvider.cs

```diff
@@ -0,0 +1,416 @@
+            }
+            else if (diagnosticId == CS1503
+                && targetNode.GetAncestorsOrThis<ArgumentSyntax>().FirstOrDefault() is ArgumentSyntax targetArgument
+                && targetArgument.Parent is ArgumentListSyntax argumentList
+                && argumentList.Parent is SyntaxNode invocationNode) // invocation node could be Invocation Expression, Object Creation, Base Constructor...
+            {
+                // Implicit downcast appears on the argument of invocation node, get all candidate functions and extract potential conversion types 
+                var symbolInfo = semanticModel.GetSymbolInfo(invocationNode, cancellationToken);
+                var candidateSymbols = symbolInfo.CandidateSymbols;
+
+                foreach (var candidateSymbol in candidateSymbols.OfType<IMethodSymbol>())
+                {
+                    if (CanArgumentTypesBeConvertedToParameterTypes(semanticModel, argumentList.Arguments,
+                        candidateSymbol.Parameters, targetArgument, cancellationToken, out var paramIndex))
+                    {
+                        var correspondingParameter = candidateSymbol.Parameters[paramIndex];
+                        var argumentConversionType = correspondingParameter.Type;
+
+                        if (correspondingParameter.IsParams
+                            && correspondingParameter.Type is IArrayTypeSymbol arrayType
+                            && !(targetNodeType is IArrayTypeSymbol))
+                        {
+                            // target argument is matched to the parameter with keyword params
+                            argumentConversionType = arrayType.ElementType;
+                        }
+
+                        mutablePotentialConversionTypes.Add(argumentConversionType);
+                    }
+                }
+
+                // Sort the potential conversion types by inheritance distance, so that
+                // operations are in order and user can choose least specific types(more accurate)
+                var comparer = new InheritanceDistanceComparer(semanticModel, targetNodeType);
+                mutablePotentialConversionTypes.Sort(comparer);
+            }
+
+            using var __ = ArrayBuilder<ITypeSymbol>.GetInstance(out var validPotentialConversionTypes);
+            foreach (var targetNodeConversionType in mutablePotentialConversionTypes)
+            {
+                var commonConversion = semanticModel.Compilation.ClassifyCommonConversion(
+                    targetNodeType, targetNodeConversionType);
+
+                // For cases like object creation expression. for example:
+                // Derived d = [||]new Base();
+                // It is always invalid except the target node has explicit conversion operator or is numeric.
+                if (targetNode.IsKind(SyntaxKind.ObjectCreationExpression)
+                    && !(commonConversion.IsUserDefined || commonConversion.IsNumeric))
```

Comment:
the `numeric` case seems like something we shouldn't have to care about :)

Reply:
Here the case is like `int i = [||]new Double();`. Since `int i = (int)new Double();` is valid, I don't know whether I need to keep it or not

Reply:
yeah, i would just dump that case.  it's not a realistic user scenario.  `new numeric_type` is just not something that happens enough to warrant special handling here :)